presto-main/src/main/java/com/facebook/presto/operator/OperatorInfoUnion.java

```diff
@@ -0,0 +1,258 @@
+
+    @ThriftConstructor
+    public OperatorInfoUnion(JoinOperatorInfo joinOperatorInfo)
+    {
+        this.joinOperatorInfo = joinOperatorInfo;
+        this.id = 7;
+    }
+
+    @ThriftField(value = 7, requiredness = OPTIONAL)
+    public JoinOperatorInfo getJoinOperatorInfo()
+    {
+        return joinOperatorInfo;
+    }
+
+    @ThriftConstructor
+    public OperatorInfoUnion(WindowInfo windowInfo)
+    {
+        this.windowInfo = windowInfo;
+        this.id = 8;
+    }
+
+    @ThriftField(value = 8, requiredness = OPTIONAL)
+    public WindowInfo getWindowInfo()
+    {
+        return windowInfo;
+    }
+
+    @ThriftConstructor
+    public OperatorInfoUnion(TableWriterOperator.TableWriterInfo tableWriterInfo)
+    {
+        this.tableWriterInfo = tableWriterInfo;
+        this.id = 9;
+    }
+
+    @ThriftField(value = 9, requiredness = OPTIONAL)
+    public TableWriterOperator.TableWriterInfo getTableWriterInfo()
+    {
+        return tableWriterInfo;
+    }
+
+    public OperatorInfoUnion(TableWriterMergeInfo tableWriterMergeInfo)
+    {
+        this.tableWriterMergeInfo = tableWriterMergeInfo;
+        this.id = 10;
+    }
+
+    @ThriftField(value = 10, requiredness = OPTIONAL)
+    public TableWriterMergeInfo getTableWriterMergeInfo()
+    {
+        return tableWriterMergeInfo;
+    }
+
+    @ThriftUnionId
+    public short getId()
+    {
+        return id;
+    }
+
+    public static OperatorInfoUnion convertToOperatorInfoUnion(OperatorInfo info)
+    {
+        if (info instanceof ExchangeClientStatus) {
+            return new OperatorInfoUnion((ExchangeClientStatus) info);
+        }
+        else if (info instanceof LocalExchangeBufferInfo) {
+            return new OperatorInfoUnion((LocalExchangeBufferInfo) info);
+        }
+        else if (info instanceof TableFinishInfo) {
+            return new OperatorInfoUnion((TableFinishInfo) info);
+        }
+        else if (info instanceof SplitOperatorInfo) {
+            return new OperatorInfoUnion((SplitOperatorInfo) info);
+        }
+        else if (info instanceof HashCollisionsInfo) {
+            return new OperatorInfoUnion((HashCollisionsInfo) info);
+        }
+        else if (info instanceof PartitionedOutputInfo) {
+            return new OperatorInfoUnion((PartitionedOutputInfo) info);
+        }
+        else if (info instanceof JoinOperatorInfo) {
+            return new OperatorInfoUnion((JoinOperatorInfo) info);
+        }
+        else if (info instanceof WindowInfo) {
+            return new OperatorInfoUnion((WindowInfo) info);
+        }
+        else if (info instanceof TableWriterOperator.TableWriterInfo) {
+            return new OperatorInfoUnion((TableWriterOperator.TableWriterInfo) info);
+        }
+        else if (info instanceof TableWriterMergeInfo) {
+            return new OperatorInfoUnion((TableWriterMergeInfo) info);
+        }
+        else {
+            throw new IllegalArgumentException("OperatorInfo is of an unknown type");
```

Comment:
Perhaps also print the classname of the unknown type?  Ditto below

Reply:
done